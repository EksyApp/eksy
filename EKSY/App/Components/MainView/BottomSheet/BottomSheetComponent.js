import React, {Component} from 'react'
import {View, Animated, StyleSheet, Dimensions} from 'react-native'
import {Badge} from "react-native-elements";
import MarkerCarousel from "./MarkerCarousel";
import Interactable from 'react-native-interactable'
import * as Theme from "../../../Theme";
import PropTypes from 'prop-types'
import {MarkersShape} from "../../../Utils/PropTypeShapes";
import RouteInfoContainer from '../../../Containers/MainView/BottomSheet/RouteInfoContainer'

const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

//Renders the bottom sheet  view
//Holds marker carousel (MarkerCarousel) containing marker cards (MarkerCard)
class BottomSheetComponent extends Component {

	constructor(props) {
		super(props)
		this._deltaY = new Animated.Value(Screen.height - 100);
	}
	
	renderContent() {
		if(this.props.routeIsActive) {
			return(
					<RouteInfoContainer/>
			)
		} else {
			return (
					<MarkerCarousel
							setMarkerSelected={this.props.setMarkerSelected}
							setMarkerViewVisible={this.props.setMarkerViewVisible}
							disableGestures={this.props.disableGestures}
							markerList={this.props.markerList}
							pointerEvents="none"
					/>
			)
		}
	}

	render() {
		return (
				<View style={styles.container}>
					<Animated.View
							style={[styles.container, {
								backgroundColor: 'black',
								opacity: this._deltaY.interpolate({
									inputRange: [0, Screen.height - 100],
									outputRange: [1, 0],
									extrapolateRight: 'clamp'
								})
							}]}
							pointerEvents="none"/>
					<Interactable.View
							verticalOnly={true}
							snapPoints={[{y: Screen.height - 220}, {y: Screen.height}]}
							boundaries={{top: 300}}
							initialPosition={{y: Screen.height}}
							animatedValueY={this._deltaY}>
						<Animated.View style={styles.panel}>
							<Animated.View style={styles.panelHeader}>
								<Animated.View style={styles.panelHandle}/>
								{this.props.markerList && this.props.markerList.length > 0 &&
								<Badge
										value={this.props.markerList.length}
										containerStyle={styles.badgeContainer}
										textStyle={styles.badgeText}
								/>
								}
							</Animated.View>
							<View style={styles.panelVisible}>
								{this.renderContent()}
							</View>

						</Animated.View>
					</Interactable.View>
				</View>
		)
	}
}

const styles = StyleSheet.create({

	container: {
		position: 'absolute',
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
		flex: 1,
	},
	viewContainer: {
		flex: 1,
	},
	panel: {
		width: Screen.width,
		height: Screen.height,
	},

	panelVisible: {
		paddingTop: 20,
		width: "100%",
		height: Screen.height,
		backgroundColor: Theme.backgroundColor,
		borderTopLeftRadius: 20,
		borderTopRightRadius: 20,
		shadowColor: '#000000',
		shadowOffset: {width: 0, height: 0},
		shadowRadius: 5,
		shadowOpacity: 0.4
	},

	panelHeader: {
		alignItems: 'center',
		position: 'relative',
		height: 20
	},
	panelHandle: {
		width: 40,
		height: 8,
		borderRadius: 4,
		backgroundColor: Theme.detailColor,
		top: 30,
		zIndex: 1,
		position: "absolute"
	},
	badgeContainer: {
		backgroundColor: '#800000',
		position: 'relative',
		top: 22,
		left: 50
	},
	badgeText: {
		color: '#FFFFFF'
	}
})

BottomSheetComponent.propTypes = {
	markerList: MarkersShape,
	setMarkerSelected: PropTypes.func,
	setMarkerViewVisible: PropTypes.func,
	disableGestures: PropTypes.func,
	routeIsActive: PropTypes.bool
}

export default BottomSheetComponent
