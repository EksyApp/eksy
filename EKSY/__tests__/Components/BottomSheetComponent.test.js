import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BottomSheetComponent from "../../App/Components/MainView/BottomSheet/BottomSheetComponent";

jest.mock('../../App/Containers/MainView/BottomSheet/RouteInfo/RouteInfoContainer', () => {
	return require('react-native').View
})

jest.mock('../../App/Containers/MainView/BottomSheet/MarkersInfo/MarkersInfoContainer', () => {
	return require('react-native').View
})

jest.mock('react-native-snap-carousel', () => {
	const React = require.requireActual('react');
	const {View} = require('react-native')
	
	class MockCarousel extends React.Component {
		
		render() {
			return (<View {...this.props} />);
		}
	}
	MockCarousel.style = {}
	return MockCarousel
})

test('BottomSheetComponent renders correctly', () => {
	const tree = renderer.create(
			<BottomSheetComponent
					markerList={[]}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});