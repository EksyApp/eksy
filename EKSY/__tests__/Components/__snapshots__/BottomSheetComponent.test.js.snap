// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheetComponent renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "flex": 1,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <InteractableView
    animatedValueX={undefined}
    animatedValueY={undefined}
    boundaries={
      Object {
        "top": 300,
      }
    }
    dragToss={0.1}
    initialPosition={
      Object {
        "y": 1259,
      }
    }
    onAnimatedEvent={[Function]}
    reportOnAnimatedEvents={true}
    snapPoints={
      Array [
        Object {
          "y": 1039,
        },
        Object {
          "y": 1259,
        },
      ]
    }
    verticalOnly={true}
  >
    <View
      style={
        Object {
          "height": 1259,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 20,
            "position": "relative",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgb(112,112,112)",
              "borderRadius": 4,
              "height": 8,
              "position": "absolute",
              "top": 30,
              "width": 40,
              "zIndex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgb(243,244,229)",
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "height": 1259,
            "paddingTop": 20,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 5,
            "width": "100%",
          }
        }
      >
        <View />
      </View>
    </View>
  </InteractableView>
</View>
`;
